app/src/main/java/co/co/nimblehq/showcases/poly/place/SliceProvider.kt

```diff
@@ -0,0 +1,101 @@
+
+class SliceProvider : SliceProvider() {
+    /**
+     * Instantiate any required objects. Return true if the provider was successfully created,
+     * false otherwise.
+     */
+    override fun onCreateSliceProvider(): Boolean {
+        return true
+    }
+
+    /**
+     * Converts URL to content URI (i.e. content://co.co.nimblehq.showcases.poly.place...)
+     */
+    override fun onMapIntentToUri(intent: Intent?): Uri {
+        // Note: implementing this is only required if you plan on catching URL requests.
+        // This is an example solution.
+        var uriBuilder: Uri.Builder = Uri.Builder().scheme(ContentResolver.SCHEME_CONTENT)
+        if (intent == null) return uriBuilder.build()
+        val data = intent.data
+        val dataPath = data?.path
+        if (data != null && dataPath != null) {
+            val path = dataPath.replace("/", "")
+            uriBuilder = uriBuilder.path(path)
+        }
+        val context = context
+        if (context != null) {
+            uriBuilder = uriBuilder.authority(context.packageName)
+        }
+        return uriBuilder.build()
+    }
+
+    /**
+     * Construct the Slice and bind data if available.
+     */
+    override fun onBindSlice(sliceUri: Uri): Slice? {
+        // Note: you should switch your build.gradle dependency to
+        // slice-builders-ktx for a nicer interface in Kotlin.
+        val context = context ?: return null
+        val activityAction = createActivityAction() ?: return null
+        return if (sliceUri.path == "/") {
+            // Path recognized. Customize the Slice using the androidx.slice.builders API.
+            // Note: ANR and StrictMode are enforced here so don"t do any heavy operations. 
+            // Only bind data that is currently available in memory.
+            ListBuilder(context, sliceUri, ListBuilder.INFINITY)
+                .addRow(
+                    ListBuilder.RowBuilder()
+                        .setTitle("URI found.")
+                        .setPrimaryAction(activityAction)
+                )
+                .build()
+        } else {
+            // Error: Path not found.
+            ListBuilder(context, sliceUri, ListBuilder.INFINITY)
+                .addRow(
+                    ListBuilder.RowBuilder()
+                        .setTitle("URI not found.")
+                        .setPrimaryAction(activityAction)
+                )
+                .build()
+        }
+    }
+
+    private fun createActivityAction(): SliceAction? {
+        return null
+        /*
+        Instead of returning null, you should create a SliceAction. Here is an example:
+        return SliceAction.create(
+            PendingIntent.getActivity(
+                context, 0, Intent(context, MyActivityClass::class.java), 0
+            ),
+            IconCompat.createWithResource(context, R.drawable.ic_launcher_foreground),
+            ListBuilder.ICON_IMAGE,
+            "Open App"
+        )
+        */
+    }
```

Comment:
Ditto here 😄 and I think we should follow [the comment pattern](https://nimblehq.co/compass/development/code-conventions/kotlin/#comments) as using `/**`
```suggestion
    /**
     * Instead of returning null, you should create a SliceAction. Here is an example:
     *    return SliceAction.create(
     *         PendingIntent.getActivity(
     *         context, 0, Intent(context, MyActivityClass::class.java), 0
     *         ),
     *         IconCompat.createWithResource(context, R.drawable.ic_launcher_foreground),
     *         ListBuilder.ICON_IMAGE,
     *         "Open App"
     *         )
     */
    private fun createActivityAction(): SliceAction? = null
```